salt/modules/ssh.py

```diff
@@ -149,15 +149,8 @@ def _replace_auth_key(
                     # Commented Line
                     lines.append(line)
                     continue
-                comps = line.split()
-                if len(comps) < 2:
-                    # Not a valid line
-                    lines.append(line)
-                    continue
-                key_ind = 1
-                if comps[0][:4:] not in ['ssh-', 'ecds']:
-                    key_ind = 2
-                if comps[key_ind] == key:
+                comps = re.findall(r'((.*)\s)?(ssh-[a-z0-9-]+|ecdsa-[a-z0-9-]+)\s([a-zA-Z0-9+/]+={0,2})(\s(.*))?', line)
```

Comment:
I'd say that compared to previous solution it's a little more precise. Thoug, I think the 'unique line id', if I may call it so, is the key itself.
I want salt (though I may not be average salt user) to see, if there is such a key in the file, and if so, make sure it has the options, type and comment I asked it to have. 
From my point of view, any key with double dash or ecdsa---foo---bar---baz, or pretty much anything else gets fixed this way.
This part of the expression really serves rather as a delimiter to figure out which part of the line really is the key. Though truth be told, thinking about it now, this could probably be done better, as having ssh-rsa somewhere in the options, would confuse the regexp as well.